pkg/reconciler/subscription/controller.go

```diff
@@ -44,10 +46,15 @@ func NewController(
 	subscriptionInformer := subscription.Get(ctx)
 	channelInformer := channel.Get(ctx)
 
+	featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"))
+	featureStore.WatchConfigs(cmw)
+
 	r := &Reconciler{
 		dynamicClientSet:   dynamicclient.Get(ctx),
+		crdLister:          customresourcedefinition.Get(ctx).Lister(),
```

Comment:
I would feel better if we injected a KRef resolver object into the reconciler, to allow us to replace it with the ClusterDuckType style resolution in the near future. And then we don't have to inject the CRD Lister into the reconciler. The interface for KRef would let you convert it to something like KRefWithVersion or something along those lines, or a GVK / GVR...

Reply:
Something like that? https://github.com/knative/pkg/pull/2127/commits/7303d0bb972f10418fb682e510a8058530cdc45b